upup/pkg/fi/cloudup/vspheretasks/attachiso.go

```diff
@@ -109,32 +116,55 @@ func createUserData(startupStr string, dir string) {
 	}
 	startupStr = strings.Join(strArray, "\n")
 
+	dnsURL, err := url.Parse(dnsServer)
+	if err != nil {
+		return err
+	}
+	dnsHost, _, err := net.SplitHostPort(dnsURL.Host)
+	if err != nil {
+		return err
+	}
+	var lines []string
+	lines = append(lines, "       echo \"nameserver "+dnsHost+"\" >> /etc/resolvconf/resolv.conf.d/head")
+	lines = append(lines, "       resolvconf -u")
+	dnsUpdateStr := strings.Join(lines, "\n")
+
 	data := strings.Replace(userDataTemplate, "$SCRIPT", startupStr, -1)
```

Comment:
I would recommend moving this to line#118 so that it's clear that startupStr is for $SCRIPT and dnsUpdateStr is for $DNS_SCRIPT. Also variables will be closer to their usage.